### PR TITLE
[asio] Update to 1.16.1

### PR DESCRIPTION
**Pull request by podsvirov**

### CODE REVIEW

File: ports/asio/CONTROL [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Source: asio
-Version: 1.12.2-2
+Version: 1.16.1-1
```
Comment (c72578) [inline ports/asio/CONTROL:2]: Should be:
`Version: 1.16.1`

Comment (podsvirov) [reply ports/asio/CONTROL]: Done.

**Comment by c72578** [comment]
Please remove `include(vcpkg_common_functions)`, it is deprecated.

**Comment by podsvirov** [comment]
> Please remove include(vcpkg_common_functions), it is deprecated.

Done.

**Comment by LilyWangL** [comment]
Thanks for your PR. Ports `restbed` and `opendnp3` depend on `asio`, now they build failed due to asio update to 1.16.1. Could you continue to fix this PR? You can click **Details** to get more information form CI test.

**Comment by LilyWangL** [comment]
@podsvirov Could you continue to fix this PR?

**Comment by podsvirov** [comment]
Looks like the [restbed](https://github.com/Corvusoft/restbed) still not ready for such a change. I created [ISSUE](https://github.com/Corvusoft/restbed/issues/431) to discuss this topic.

**Comment by podsvirov** [comment]
Looks like the [opendnp3](https://github.com/dnp3/opendnp3) still not ready for such a change. I created [ISSUE](https://github.com/dnp3/opendnp3/issues/373) to discuss this topic.

**Comment by BillyONeal** [comment]
@podsvirov It looks like https://github.com/dnp3/opendnp3/issues/373 is now fixed, are you still interested in this?

**Comment by podsvirov** [comment]
Hello @BillyONeal, I'm here. Interest has not faded yet :-)
Are you suggesting to update the associated ports in this PR?

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**